docs/_content/ontwikkelaars/algemeen/audit-trail.md

```diff
@@ -0,0 +1,248 @@
+status code van "200" of "201" is bijvoorbeeld een succesvol uitgevoerde 
+handeling, terwijl een "403" aangeeft dat de gebruiker of applicatie het wel
+geprobeerd heeft maar de handeling is niet uitgevoerd.
+
+`hoofdObject` (verplicht)
+
+De URL naar het hoofd object van een component *zoals bij notificaties*. Bij 
+het ZRC is dit dus altijd de URL naar een zaak.
+
+`resource` (verplicht)
+
+De resource naam *zoals bij notificaties*.
+
+`resourceUrl` (verplicht)
+
+De URL naar het object *zoals bij notificaties*.
+
+`resourceWeergave` (optioneel)
+
+Vriendelijke identificatie van het object. Bijvoorbeeld de zaak identificatie 
+of de omschrijving van een status.
+
+`toelichting` (optioneel)
+
+Toelichting waarom de handeling is uitgevoerd. Dit kan handig zijn voor sommige
+handelingen waarbij additionele context gewenst is, zoals het ophogen van een
+archiveringstermijn of een correctie buiten het reguliere proces om.
+
+`aanmaakdatum` (automatisch)
+
+De datum waarom de handeling is gedaan.
+
+#### TODO: Extra attributen
+
+In overweging om op te nemen:
+
+* `verschillen` -- Een lijst met de wijzigingen.
+* `weergave` -- De audit trail regel op vriendelijke wijze samengesteld.
+
+#### Voorbeeld
+
+```http
+GET /api/v1/zrc/audittrail/
+[
+  {
+    "uuid": "311cf2",
+    "bron": "ZRC",
+    "applicatieId": "demo-app",
+    "applicatieWeergave": "Demo applicatie",
+    "gebruikersId": "14",
+    "gebruikersWeergave": "Joeri Bekker",
+    "actie": "create",
+    "actieWeergave": "",
+    "resultaat": 201,
+    "hoofdObject": "https://www.example.com/api/v1/zaken/5ab6e2",
+    "resource": "status",
+    "resourceUrl": "https://www.example.com/api/v1/statussen/11cb71",
+    "resourceWeergave": "Ingediend",
+    "toelichting": "",
+    "aanmaakdatum": "2019-05-05T11:53:18.090384Z",
+  }
+]
+```
+
+Hiermee kan de volgende (eenvoudige) audit regel worden opgesteld:
+
+```
+<aanmaakdatum> <resource> "<resourceWeergave>" is <resultaat> <actie/actieWeergave> in <bron> door <gebruikersWeergave> (<gebruikersId>) via <applicatieWeergave> (<applicatieId)
+```
+
+Dat als instantie vertaald kan worden naar:
+
+```
+5 mei 2019 11:53 Status "Ingediend" is succesvol aangemaakt in ZRC door Joeri Bekker (14) via Demo Applicatie (demo-app).
+```
+
+### Query parameters
+
+Er kan worden gefilterd op:
+
+* `applicatieId` (`exact` en `in`)
+* `gebruikersId` (`exact` en `in`)
+* `actie` (`exact` en `in`)
+* `resultaat` (`exact`, `in`, `lte` en `gte`)
+* `hoofdObject`
+* `resource` (`exact` en `in`)
+* `resourceUrl`
+* `aanmaakdatum` (`exact`, `lte` en `gte`)
+
+En gesorteerd worden op:
+
+* `aanmaakdatum` (`asc` en `desc`)
+
+#### Voorbeelden
+
+Zo kan bijvoorbeeld een audit trail worden opgevraagd van een bepaalde zaak:
+
+```http
+GET /api/v1/zrc/audittrail?
+  hoofdObject = https://www.example.com/api/v1/zaken/5ab6e2
+```
+
+Of, om alle foutmeldingen in een bepaalde periode, door een bepaald persoon 
```

Comment:
Ik zou dit geen foutmeldingen noemen. De audittrail is niet bedoeld voor foutmeldingen maar voor reconstructie over wat er met een zaak gebeurd is. 
Zaken rond foutmeldingen vallen denk ik eerder onder logging

Reply:
Ik had eerder naar punt 4 van de NORA definitie gewezen naar Hugo toe. Maar als jullie beide vinden dat het niet hoeft, dat laat ik het weg.

Reply:
hoeft niet inderdaad.